libs/utils/src/vec_map.rs

```diff
@@ -100,40 +125,65 @@ impl<K: Ord, V> VecMap<K, V> {
         V: Clone,
     {
         let split_idx = self
-            .0
+            .data
             .binary_search_by_key(&cutoff, extract_key)
             .unwrap_or_else(std::convert::identity);
 
         (
-            VecMap(self.0[..split_idx].to_vec()),
-            VecMap(self.0[split_idx..].to_vec()),
+            VecMap {
+                data: self.data[..split_idx].to_vec(),
+                ordering: self.ordering,
+            },
+            VecMap {
+                data: self.data[split_idx..].to_vec(),
+                ordering: self.ordering,
+            },
         )
     }
 
     /// Move items from `other` to the end of `self`, leaving `other` empty.
     /// If any keys in `other` is less than or equal to any key in `self`,
     /// `InvalidKey` error will be returned and no mutation will occur.
-    pub fn extend(&mut self, other: &mut Self) -> Result<usize, InvalidKey> {
-        let self_last_opt = self.0.last().map(extract_key);
-        let other_first_opt = other.0.last().map(extract_key);
+    pub fn extend(&mut self, other: &mut Self) -> Result<usize, VecMapError> {
+        if self.ordering == VecMapOrdering::Greater
+            && other.ordering == VecMapOrdering::GreaterOrEqual
+        {
+            return Err(VecMapError::ExtendOrderingError);
+        }
```

Comment:
Could you do `if self.ordering != other.ordering` instead? This will still work if new orderings are introduced.

Reply:
Ah, I see from the test that you to allow extending a vec map ordered by `GreaterOrEqual` with a vec map ordered by `Greater`. I don't think we need this feature, so let's not add it now.